pkg/cbpf/st_ld_instructions_test.go

```diff
@@ -0,0 +1,226 @@
+			instruction:  LdIndH(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLd),
+			wantSize:     int32(pb.StLdSize_StLdSizeH),
+			wantMode:     int32(pb.StLdMode_StLdModeIND),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding LDIndB Instruction",
+			instruction:  LdIndB(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLd),
+			wantSize:     int32(pb.StLdSize_StLdSizeB),
+			wantMode:     int32(pb.StLdMode_StLdModeIND),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding LDx Instruction",
+			instruction:  Ldx(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLdx),
+			wantSize:     int32(pb.StLdSize_StLdSizeW),
+			wantMode:     int32(pb.StLdMode_StLdModeMEM),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding LDxI Instruction",
+			instruction:  Ldxi(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLdx),
+			wantSize:     int32(pb.StLdSize_StLdSizeW),
+			wantMode:     int32(pb.StLdMode_StLdModeIMM),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding LDxLen Instruction",
+			instruction:  LdxLen(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLdx),
+			wantSize:     int32(pb.StLdSize_StLdSizeW),
+			wantMode:     int32(pb.StLdMode_StLdModeLEN),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding LDxAbs Instruction",
+			instruction:  LdxAbs(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLdx),
+			wantSize:     int32(pb.StLdSize_StLdSizeW),
+			wantMode:     int32(pb.StLdMode_StLdModeABS),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding LDxb Instruction",
+			instruction:  Ldxb(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassLdx),
+			wantSize:     int32(pb.StLdSize_StLdSizeB),
+			wantMode:     int32(pb.StLdMode_StLdModeMSH),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding ST Instruction",
+			instruction:  St(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassSt),
+			wantSize:     int32(pb.StLdSize_StLdSizeW),
+			wantMode:     int32(pb.StLdMode_StLdModeMEM),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+		{
+			testName:     "Encoding STx Instruction",
+			instruction:  Stx(1, 2, testK),
+			wantClass:    int32(pb.InsClass_InsClassStx),
+			wantSize:     int32(pb.StLdSize_StLdSizeW),
+			wantMode:     int32(pb.StLdMode_StLdModeMEM),
+			wantJmpTrue:  1,
+			wantJmpFalse: 2,
+			wantK:        testK,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(t *testing.T) {
+			instruction := tc.instruction
+			t.Logf("Running test case %s", tc.testName)
+
+			if (instruction.Opcode & 0x07) != tc.wantClass {
```

Comment:
same comment for the constants. Would it be helpful to declare them in constants.go?